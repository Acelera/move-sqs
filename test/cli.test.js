const moveSqs = require('../src/move-sqs');

jest.mock('../src/move-sqs');

describe('cli', () => {
  it('should parse the command line args and invoke the move-sqs function', () => {
    process.argv = [
      'node',
      'cli',
      '--source-queue-url',
      'http://the.source-queue-url.com',
      '--destination-queue-url',
      'http://the.destination-queue-url.com',
      '--region',
      'eu-west-1',
      '--access-key-id',
      'theAccessKey',
      '--secret-access-key',
      'theSecretAccessKey',
      'merge-json-message-with',
      JSON.stringify({ timeout: '2019-02-01T12:00' }),
    ];

    const cli = require('../src/cli'); // eslint-disable-line global-require
    expect(cli.flags).toMatchSnapshot();
    expect(moveSqs).toHaveBeenCalledWith(cli.flags);
  });
});
