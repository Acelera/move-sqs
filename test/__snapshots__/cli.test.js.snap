// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should parse the command line args and invoke the move-sqs function 1`] = `
Object {
  "accessKeyId": "theAccessKey",
  "destinationQueueUrl": "http://the.destination-queue-url.com",
  "region": "eu-west-1",
  "secretAccessKey": "theSecretAccessKey",
  "sourceQueueUrl": "http://the.source-queue-url.com",
}
`;
